### PR TITLE
Initial server

### PR DESCRIPTION
**Pull request by DrPandemic**
- Adds a small interface for bitcoin-core.
- Adds an experimental interface for getBlockTemplate's template.
- Updates the gulpfile

Currently, the main file connects to the bitcoind and ask for a template.

### CODE REVIEW

File: server/typings/tsd.d.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+/// <reference path="node/node.d.ts" />
```
Comment (JesseEmond) [inline server/typings/tsd.d.ts]: Should this be versionned? Idem for `server/typings/node/node.d.ts` (which is relatively large).

File: server/tsd.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "version": "v4",
+  "repo": "borisyankov/DefinitelyTyped",
```
Comment (JesseEmond) [inline server/tsd.json]: Should we update this to `DrPandemic/pickaxe`?

**Comment by JesseEmond** [comment]
Minor comments, LGTM :+1: after.

File: server/tsd.json (hunk 2 above)
Comment (DrPandemic) [inline server/tsd.json]: Nope. I think it represent the source of the files. But 14 days ago, in the repo `DefinitelyTyped/DefinitelyTyped` all those links seems to have changed to `DefinitelyTyped/DefinitelyTyped`

**Pull request merged**